app/ide-desktop/lib/dashboard/src/components/dashboard/Permission.tsx

```diff
@@ -38,7 +38,10 @@ const ASSET_TYPE_TO_TEXT_ID: Readonly<Record<backendModule.AssetType, text.TextI
 /** Props for a {@link Permission}. */
 export interface PermissionProps {
   readonly backend: Backend
-  readonly asset: backendModule.Asset
+  readonly asset: {
```

Comment:
a bit awkward imo, i think it's better if it's flattened. like i'd understand if this was perhaps an existing type, but as it is you specify it as `asset={{ id, type }}` which just feels like a weird mix of jsx and regular js

Reply:
> specify it as asset={{ id, type }} which just feels like a weird mix of jsx and regular js

Yeah, but I introduced this type to declare exactly which fields it expects as an input. We still can pass assets as is. IMO we'd use a format like that `asset={{ id, type }}` only inside tests, and it's good IMO because we don't need to maintain unneeded fields in mocks.

Reply:
see other comment... somewhere else then - perhaps `Pick` might be a better idea? for multiple reasons:
- to let the user know "this is a subset of `AnyAsset`, so you can pass an `AnyAsset` here
- so that it never goes out of sync with `AnyAsset`